schema/schema.go

```diff
@@ -187,8 +190,76 @@ func (s *Schema) ValidateData(data []byte) error {
 			return errors.Wrap(err, "failed to JSON remarshal data for validation")
 		}
 	}
+	if err = s.validate(schema.NewBytesLoader(data)); err != nil {
+		return err
+	}
+	return s.validateSpecAnnotations(data)
```

Comment:
I would personally insert this before the call to `s.validate`. I would also in this case unconditionally call `yaml.Unmarshal` and pass `any` to the annotation validation methods below.

Reply:
Why before? I want to have a valid document before I go and check the annotations.

Reply:
Sorry, I mean, there is a conditional `yaml.Unmarshal` call to heck whether we can re-encode the YAML as JSON if required. I was suggesting we reuse this unmarshalled object for our annotation validations instead of re-Unmarshalling the data there.

Reply:
Ah right, thanks for the point!! PTAL